src/main/java/io/r2dbc/mssql/message/tds/Decode.java

```diff
@@ -224,4 +224,35 @@ private static String decodeUnicode(ByteBuf buffer, int length) {
 
         return result;
     }
+
+    /**
+     * Decode a {@link RoutingData} from {@link ByteBuf}.
+     *
+     * @param buffer the data buffer
+     * @return the decoded {@link RoutingData}
+     */
+    public static RoutingData decodeRoute(ByteBuf buffer) {
```

Comment:
Let's move decoding into `RoutingData` itself (`public static RoutingData decode(ByteBuf buffer) {`). `Decode` is intended for low-level primitives.

Reply:
Done